src/jit/morph.cpp

```diff
@@ -11221,15 +11221,20 @@ CM_OVF_OP :
             }
             else
             {
-                /* The left operand is worthless, throw it away */
-                if (lvaLocalVarRefCounted)
+                // If this is an intrinsic that represents the object.GetType(), it can throw an NullReferenceException.
+                // Don't remove such call.
```

Comment:
Morph logic doesn't need to special case any intrinsic like this once OperMayThrow() is properly defined for GetType() intrinsic.  Please address this before the check-in.